api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -96,7 +88,7 @@ message RateLimit {
   // the `draft RFC <https://tools.ietf.org/id/draft-polli-ratelimit-headers-03.html>`_.
   //
   // Disabled by default.
-  XRateLimitHeadersRFCVersion enable_x_ratelimit_headers = 8
+  common.ratelimit.v3.XRateLimitHeadersRFCVersion enable_x_ratelimit_headers = 8
```

Comment:
Unfortunately this breaks backwards compatibility, and therefore the field must be deprecated and a new field with the new type should be introduced.

@htuch extracting the enum to a common place (see: `api/envoy/extensions/common/ratelimit/v3/ratelimit.proto`) is arguably the right way to go, but will cause some deprecation churn. Another option is to make the local_rate_limit proto use the enum directly from the (global) rate_limit proto. WDYT?

Reply:
I think for the local rate limiting, put it in the common location and use it there. Either deprecate or eave a TODO to cleanup for global. I'm not super keen on API dependencies between extensions, since we're creating an unnecessary dependency in build/protos.

Reply:
In my opinion, if we treat local rate limit and rate limit as two independent extensions (like if we can decouple envoy and extensions someday in the future), both local rate limit and rate limit should maintain their own enum values. But if we consider the local rate limit and rate limit are inseparable, extract common enum is a good choice.

Reply:
My take on this is that similar to the original issue (#15293) that requested a local rate-limiting feature support similar to what the global rate-limiting does, then in the future they should be in-sync.
I think that adding the enum to the common proto (as you've already done) is a good thing, and to avoid changes to the global rate-limiting API (following @htuch suggestion), we should not deprecate this enum at this stage, but add a `[#next-major-version:` comment (see: example [here](https://github.com/envoyproxy/envoy/blob/9d3cf43e3fab618dd2bedd0deb2df973760bbbd6/api/envoy/type/matcher/v3/metadata.proto#L77)).